app/models/user.rb

```diff
@@ -116,10 +116,10 @@ def dummy_password_for_external_auth
   def change_password(oldpwd, newpwd)
     auth = self.class.authenticator(userid)
     raise MiqException::MiqEVMLoginError, "password change not allowed when authentication mode is #{auth.class.proper_name}" unless auth.uses_stored_password?
-    raise MiqException::MiqEVMLoginError, "old password does not match current password" unless User.authenticate(userid, oldpwd)
-
-    self.password = newpwd
-    self.save!
+    if User.authenticate(userid, oldpwd)
```

Comment:
To answer your question more directly, this is what is passed to the user (at least on the password change screen):

[Error here](https://github.com/manageiq/manageiq/blob/master/app/services/user_validation_service.rb#L103) is [passed to the controller](https://github.com/manageiq/manageiq/blob/master/app/controllers/dashboard_controller.rb#L504-L521)